backend/src/middleware/passportStrategies.ts

```diff
@@ -57,9 +57,15 @@ export const jwtStrategy = new JwtStrategy(
     secretOrKey: process.env.JWT_SECRET_KEY,
     jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
   },
-  (token, done) => {
+  async (token, done) => {
     try {
-      return done(null, token.user);
+      const user = token.user as User;
+
+      if (!(await User.count({ where: { id: user.id } }))) {
+        return done(makeError('User does not exist', status.UNAUTHORIZED), null);
+      }
+
+      return done(null, user);
```

Comment:
Just a general question. Since there is already a query being done on the database, would something like this be more future proof:
```js
const user = await User.findByPk(token.user.id);
if (!user) return done(makeError('User does not exist', status.UNAUTHORIZED), null);
return done (null, user);
```
I assume TypeScript will have some issues with this but the general idea is to. have the full model of the user available on the request. So if there is a need for specific user field being available it is already there.

Reply:
something like this does not seem to work. for now, we will stick to previous implementation because we use only id from request user.

Reply:
Which specific part does not work? I was able to make it work but the issue after that is that the default scope excludes the `id` so that makes it now work in router handlers like delete profile.

Reply:
It works but you have to remove 'id' from exclude in user model default scope.

Reply:
that fixed it. thanks!